src/images.py

```diff
@@ -0,0 +1,177 @@
+    else:
+        bg_row_slope = np.nan
+
+    if np.any(valid_cols):
+        bg_col_slope = np.polyfit(np.arange(bg_avg_col.size)[valid_cols], bg_avg_col[valid_cols], 1)[0]
+    else:
+        bg_col_slope = np.nan
+
+    bg_mag = np.sqrt(bg_row_slope**2 + bg_col_slope**2)
+
+    # plot the results if needed
+    if plot:
+        colors = {
+            "fg_avg": "#A6CEE3",
+            "fg_slope": "#1F78B4",
+            "bg_avg": "#FDBF6F",
+            "bg_slope": "#FF7F00"
+        }
+
+        fig, axes = plt.subplots(1, 2, figsize=(12, 5))
+
+        x_rows = np.arange(fg_avg_row.size)
+        x_cols = np.arange(fg_avg_col.size)
+
+        # Row plot
+        axes[0].plot(fg_avg_row, label="Foreground Avg", color=colors["fg_avg"])
+        axes[0].plot(bg_avg_row, label="Background Avg", color=colors["bg_avg"])
+        fg_valid_rows = ~np.isnan(fg_avg_row)
+        bg_valid_rows = ~np.isnan(bg_avg_row)
+        axes[0].plot(x_rows[fg_valid_rows], fg_row_slope * x_rows[fg_valid_rows] + np.polyfit(x_rows[fg_valid_rows], fg_avg_row[fg_valid_rows], 1)[1],
+                    label="Foreground Slope", color=colors["fg_slope"], linestyle='--')
+        axes[0].plot(x_rows[bg_valid_rows], bg_row_slope * x_rows[bg_valid_rows] + np.polyfit(x_rows[bg_valid_rows], bg_avg_row[bg_valid_rows], 1)[1],
+                    label="Background Slope", color=colors["bg_slope"], linestyle='--')
+        axes[0].set_xlabel("Row")
+        axes[0].set_ylabel("Average Intensity / Slope")
+        axes[0].set_title("Row-wise Average Intensity & Slope")
+
+        # Column plot
+        axes[1].plot(fg_avg_col, label="Foreground Avg", color=colors["fg_avg"])
+        axes[1].plot(bg_avg_col, label="Background Avg", color=colors["bg_avg"])
+        fg_valid_cols = ~np.isnan(fg_avg_col)
+        bg_valid_cols = ~np.isnan(bg_avg_col)
+        axes[1].plot(x_cols[fg_valid_cols], fg_col_slope * x_cols[fg_valid_cols] + np.polyfit(x_cols[fg_valid_cols], fg_avg_col[fg_valid_cols], 1)[1],
+                    label="Foreground Slope", color=colors["fg_slope"], linestyle='--')
+        axes[1].plot(x_cols[bg_valid_cols], bg_col_slope * x_cols[bg_valid_cols] + np.polyfit(x_cols[bg_valid_cols], bg_avg_col[bg_valid_cols], 1)[1],
+                    label="Background Slope", color=colors["bg_slope"], linestyle='--')
+        axes[1].set_xlabel("Column")
+        axes[1].set_ylabel("Average Intensity / Slope")
+        axes[1].set_title("Column-wise Average Intensity & Slope")
+
+        # Collect unique handles and labels for a single legend
+        handles, labels = [], []
+        for ax in axes:
+            for h, l in zip(*ax.get_legend_handles_labels()):
+                if l not in labels:  # only add unique labels
+                    handles.append(h)
+                    labels.append(l)
+
+        fig.legend(handles, labels, loc='center left', bbox_to_anchor=(1, 0.5), fontsize=10)
+        plt.tight_layout()
+        plt.show()
```

Comment:
Would you ever want the figure returned from this function so it could be modified or have delayed display outside of this function?